### PR TITLE
Tagging - Add more tags

### PR DESCRIPTION
**Pull request by pterolatypus**
**When merged this pull request will:**
- Added 10 new tags in each colour
- Tags are now grouped by required item in the interaction menu

<details>
<summary>Screenshots</summary>

</details>

### IMPORTANT

- [x] If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [x] [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- [x] Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/tagging/ACE_Tags.hpp [hunk 1]
```diff
@@ -1,3 +1,10 @@
+#define TAG(name,col) class TRIPLES(ACE,name,col) { \
+        displayName = CSTRING(DOUBLES(name,col)); \
+        requiredItem = QUOTE(ACE_Spraypaint##col); \
```
Comment (PabstMirror) [inline addons/tagging/ACE_Tags.hpp:3]:
```suggestion        
        requiredItem = QUOTE(GLUE(ACE_Spraypaint,col)); \
```

File: addons/tagging/ACE_Tags.hpp [hunk 2]
```diff
@@ -1,3 +1,10 @@
+#define TAG(name,col) class TRIPLES(ACE,name,col) { \
```
Comment (PabstMirror) [inline addons/tagging/ACE_Tags.hpp:1]:
```suggestion
#define GLUE(g1,g2) g1##g2
#define TAG(name,col) class TRIPLES(ACE,name,col) { \
```

File: addons/tagging/functions/fnc_addTagActions.sqf [hunk 3]
```diff
@@ -41,16 +41,30 @@ private _actions = [];
                 [_unit, _randomTexture, _randomMaterial] call FUNC(tag);
                 _unit setVariable [QGVAR(lastUsedTag), _class];
             },
-            {
-                (_this select 2) params ["_unit", "", "", "_requiredItem"];
-                _requiredItem in (_unit call EFUNC(common,uniqueItems))
-            },
+            {},
```
Comment (PabstMirror) [inline addons/tagging/functions/fnc_addTagActions.sqf:44]:
```suggestion
            {true},
```

File: addons/tagging/functions/fnc_addTagActions.sqf [hunk 4]
```diff
@@ -41,16 +41,30 @@ private _actions = [];
...
+    _itemsActions set [_requiredItem, _actions];
 } forEach GVAR(cachedTags);
 
+private _actions = [];
+{
+    _actions pushBack [
+        [
+            format ["ACE_TagItem_%1", _x],
+            getText (configFile >> "CfgWeapons" >> _x >> "displayName"),
+            (_y select 0  select 0 select 2), //use icon of first child action in the list
```
Comment (PabstMirror) [inline addons/tagging/functions/fnc_addTagActions.sqf:60]: could use the item's picture
```suggestion
            getText (configFile >> "CfgWeapons" >> _x >> "picture"),
```

File: addons/tagging/ACE_Tags.hpp (hunk 1 above)
Comment (PabstMirror) [reply addons/tagging/ACE_Tags.hpp:3]: fixes 
`[ACE] (tagging) ERROR: Failed compiling ACE_Tags for tag: ACE_ar_u_Black - requiredItem ACE_Spraypaint##Black does not exist`

**Comment by PabstMirror** [comment]
I think we should make the action icons the same as the tag (make config `icon` the same as `texture`)
imho it's faster and easier to find what you want (especially since text is cut off sometimes)
Untitled

File: addons/tagging/ACE_Tags.hpp (hunk 2 above)
Comment (jonpas) [reply addons/tagging/ACE_Tags.hpp:2]: `particles` use `#define MERGE(var1,var2) var1####var2`.

**Comment by jonpas** [comment]
Agreed, colour from the name should be removed. It's already in the icon and the higher-level action point.

**Comment by PabstMirror** [comment]
This kinda takes your hash stuff and merges it with `GVAR(cachedRequiredItems)`
https://github.com/acemod/ACE3/commit/b110bc90eb2ba335f1773691fa40ff5007432cd4
makes ace_tagging_fnc_addTagActions ~8x faster
let me know if this looks ok to push (besides header)

**Comment by Drofseh** [comment]
> I've left the colour in the "X Red" et al for now, since I think just "X" looks a bit weird

I have some concerns for colour blind people, but I think as long as `Spray Paint (Red)` still has the colour then `X Red` can just be `X`

**Review by PabstMirror — APPROVED** [review]

**Pull request merged**